demos/groktunnel/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+	"sync"
+	"time"
+
+	vhost "github.com/inconshreveable/go-vhost"
+	shortuuid "github.com/lithammer/shortuuid/v3"
+	"github.com/progrium/qmux/golang/session"
+)
+
+func fatal(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	var port = flag.String("p", "9999", "server port to use")
+	var host = flag.String("h", "vcap.me", "server hostname to use")
+
+	flag.Parse()
+
+	if flag.Arg(0) != "" {
+		conn, err := net.Dial("tcp", "_new."+net.JoinHostPort(*host, *port))
+		fatal(err)
+		client := httputil.NewClientConn(conn, bufio.NewReader(conn))
+		req, err := http.NewRequest("GET", "/", nil)
+		req.Host = "_new." + net.JoinHostPort(*host, *port)
+		fatal(err)
+		client.Write(req)
+		resp, _ := client.Read(req)
```

Comment:
should this exit on errors?
```suggestion
		resp, err := client.Read(req)
		fatal(err)
```